tests/test_browser.py

```diff
@@ -3108,19 +3108,26 @@ def prep_no_SAB(self):
       </script>
     '''))
 
+  # Run a browser test both with and without wasm
+  def btest_wasm(self, *args, **kwargs):
```

Comment:
for running as both asm and wasm, see the `also_proxied` flag to btest which makes us test both regular and proxied mode. maybe we can do the same pattern for asm and wasm.

Reply:
I still don't see why we need a new method `btest_wasm`, can't be do something parallel to `also_proxied`, like `also_wasm`?

otherwise lgtm

Reply:
Or is this different because it is off for now?

Reply:
No, sorry I just hadn't gotten to that yet. Its switched now. I don't see the feature-detection mechanism you mentioned though. Since the tests run in a random order we can't just rely on the outcome of one of them to gate the rest. I thought I might just add a flag to runner.py for now.

Reply:
BTW, chrome needs a flag to enable threads (which means modifying `EMSCRIPTEN_BROWSER` in the env) so I thought maybe just another env var for testing it since we have an env var anyway. Does nightly need a separate flag, or is it just on by default there?

Reply:
Not sure about how we avoid pthreads tests if no pthreads support - @juj?

Also @juj would know about firefox nightly needing a flag for wasm threads or not as he tested it.